back-end/service.py

```diff
@@ -95,50 +133,62 @@ def updateSecret(secret, data, user, repo):
         log('debug', 'Secret updated in `{}`'.format(repo))
         return True
     else:
-        log('warning', 'Secret could not be updated in `{}`'.format(repo))
-        log('critical', 'Execution may not work properly from here.')
-        return False
+        raise ServerError('Secret could not be updated in `{}`'.format(repo),
+            HTTPStatus(500))
 
 
 def ensureSecrets(user):
 
+    """
+    Runs a check on all of user's ACTIVATED Drone repositories
+    if each of them has the secret DRONE_PRIVATE_SSH_KEY and is updated
+    with the latest key.
+
+    Initiates the installation process for secret, if it doesn't exists.
+    """
+
     repos = requests.get(
         DRONE_ADDR + "/api/user/repos",
         headers={
             'Authorization': 'Bearer {}'.format(os.environ['DRONE_TOKEN'])
             }).json()
 
-    for repo in repos:
-        if repo['active']:
+    for repo in [repo for repo in repos if repo['active']]:
+
+        secrets = requests.get(
+            DRONE_ADDR + "/api/repos/{0}/{1}/secrets".format(
+                user, repo['name']),
+            headers={'Authorization': 'Bearer {}'.format(
+                os.environ['DRONE_TOKEN'])}
+            ).json()
 
-            secrets = requests.get(
-                DRONE_ADDR + "/api/repos/{0}/{1}/secrets".format(user, repo['name']),
-                headers={'Authorization': 'Bearer {}'.format(
-                    os.environ['DRONE_TOKEN'])}
-                ).json()
+        with open(os.path.join(SSH_PATH, PRIV_KEY), 'r') as key:
 
-            with open(os.path.join(SSH_PATH, PRIV_KEY), 'r') as key:
+            for secret in secrets:
+                if secret['name'] == 'DRONE_PRIVATE_SSH_KEY':
+                    log('debug', 'Secret found in repo `{}`'.format(
+                        repo['name']))
 
-                for secret in secrets:
-                    if secret['name'] == 'DRONE_PRIVATE_SSH_KEY':
-                        log('debug', 'Secret found in repo `{}`'.format(
-                            repo['name']))
+                    updateSecret(
+                        secret=secret['name'],
+                        data=key.read(),
+                        repo=repo['name'],
+                        user=user
+                    )
+                break
```

Comment:
This will break on the first iteration whether it updates the secret or not.  It should only break once it has found a `DRONE_PRIVATE_SSH_KEY`.
```suggestion
                    break
```

Reply:
Done and pushed.